.github/workflows/ci.yaml

```diff
@@ -32,6 +33,46 @@ jobs:
         run: |
           echo "::add-matcher::.github/workflows/matchers/intentfest.json"
       - run: script/lint
-      - run: script/test
+      - name: Get changed files
+        if: github.event_name == 'pull_request'
+        id: changed-files
+        uses: tj-actions/changed-files@v42
+        with:
+          files_separator: " "
+          json: true
+
+      - name: Extract affected languages
+        if: github.event_name == 'pull_request'
+        id: extract-languages
+        run: |
+          # Check if any files outside language directories were changed
+          CHANGED_FILES='${{ steps.changed-files.outputs.all_changed_files }}'
+          if echo "$CHANGED_FILES" | grep -qvE '^(sentences|responses|tests)/'; then
+            echo "Running full tests due to changes outside language directories"
+            exit 0
+          fi
+
+          # Get unique language codes from changed files
+          LANGUAGES=$(echo "$CHANGED_FILES" | \
+            grep -oE '/(sentences|responses|tests)/([^/ ]+)/' | \
+            cut -d'/' -f3 | sort -u | tr '\n' ',' | sed 's/,$//')
+
+          if [ -n "$LANGUAGES" ]; then
+            echo "languages=$LANGUAGES" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Run tests (PR)
+        if: github.event_name == 'pull_request'
+        run: |
+          if [ -n "${{ steps.extract-languages.outputs.languages }}" ]; then
+            script/test --language ${{ steps.extract-languages.outputs.languages }}
```

Comment:
Does this arg work with multiple languages ?

Reply:
It seems to work with comma separated values: `pytest tests --language he,ru`

Reply:
Yes, code confirms that's the way to go about it https://github.com/home-assistant/intents/blob/f85c1f3c74fed771a0cdabded8f2cc8e0bface47/tests/conftest.py#L27

Reply:
one of my first CI runs wrongly matched all languages and this command worked